pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/WorkerQueryExecutor.java

```diff
@@ -69,26 +65,12 @@ public synchronized void shutDown() {
   }
 
   public void processQuery(DistributedStagePlan queryRequest, Map<String, String> requestMetadataMap,
-      ExecutorService executorService) {
+      OpChainSchedulerService scheduler) {
```

Comment:
not necessarily needed in this PR: let's change this API to directly take operator chain as input
```suggestion
  public void processQuery(OpChain opChain, OpChainSchedulerService scheduler) {
```
and make the opChain constructor outside --> this way we can early return if there's any error during opChain construct